### PR TITLE
Add write interface for retrieving Vulnerability Remediation data

### PR DESCRIPTION
**Pull request by GabrielEValenzuela**
|Related issue|
|---|
|#19046|

## Objective

This pull request introduces a write interface that enables the storage of `Vulnerability Remediation` data from a CVE5 schema to the RocksDB database. With this enhancement, the vulnerability scanner can update and query data related to security vulnerabilities and their remediation efforts more efficiently and reliably. Adding this write interface significantly improves the system's capabilities, allowing for seamless integration of RocksDB as a data source, enhancing data storage speed, and providing a more robust foundation for managing security vulnerabilities within the application. This enhancement represents a critical step towards optimizing the performance and reliability of the Vulnerability Remediation feature.

## Description
- Implemented validation logic to verify the content of the CVE5 object.
- Added error-handling mechanisms to handle invalid or incomplete configurations.
- Included unit tests to verify the accuracy of the validation process.

For a better compression of the detailed work, please check this top-level sequence diagram

<details><summary>Sequence Diagram</summary>

</details>

## Quality Assurance
A new test has been included to verify the correct behavior and backward compatibility of this solution proposal.

<details><summary>Output testing 🟢 </summary>

```bash
[----------] 4 tests from StoreRemediationsModelTest
[ RUN      ] StoreRemediationsModelTest.UpdatesWindowsRemediation
[       OK ] StoreRemediationsModelTest.UpdatesWindowsRemediation (36 ms)
[ RUN      ] StoreRemediationsModelTest.SkipsEmptyRemediations
[       OK ] StoreRemediationsModelTest.SkipsEmptyRemediations (34 ms)
[ RUN      ] StoreRemediationsModelTest.SkipsEmptyUpdates
[       OK ] StoreRemediationsModelTest.SkipsEmptyUpdates (36 ms)
[ RUN      ] StoreRemediationsModelTest.x_remediationsNotPresent
[       OK ] StoreRemediationsModelTest.x_remediationsNotPresent (36 ms)
[----------] 4 tests from StoreRemediationsModelTest (145 ms total)
```

</details>

Regarding memory, no new memory allocations were introduced in this PR using `raw_pointers`.

<details><summary>ScanBuild report</summary>

Command:

```bash
scan-build-15 -enable-checker alpha.core.CastSize -enable-checker alpha.core.CastToStruct -enable-checker alpha.core.Conversion -enable-checker alpha.core.IdenticalExpr -enable-checker alpha.core.SizeofPtr -enable-checker alpha.core.TestAfterDivZero --exclude external make TARGET=server DEBUG=1 -j$(nproc)
```

```bash

Done building server

scan-build: No bugs found.

```

</details>

<details><summary>Clang-Format</summary>

```bash
---------------------------------
PASSED: Clang-format check passed
---------------------------------
```
</details>

## Exploratory tests
Not apply

### Check summary
- [x] Style and quality of SCA.
- [x] Documentation clear.
- [x] Functionality optimizated.

### CODE REVIEW

**Comment by GabrielEValenzuela** [comment]
# About failed check

Locally is not reproducible

```bash
valgrind ctest --output-on-failure
==22334== Memcheck, a memory error detector
==22334== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==22334== Using Valgrind-3.19.0 and LibVEX; rerun with -h for copyright info
==22334== Command: ctest --output-on-failure
==22334== 
Test project /home/gvalenzuela/Documents/Work/WazuhRepositories/vulnerability-refactor-II/src/build/shared_modules/indexer_connector/tests
    Start 1: indexer_connector_unit_tests
1/1 Test #1: indexer_connector_unit_tests .....   Passed   50.10 sec

100% tests passed, 0 tests failed out of 1

Total Test time (real) =  50.39 sec
==22334== 
==22334== HEAP SUMMARY:
==22334==     in use at exit: 0 bytes in 0 blocks
==22334==   total heap usage: 4,677 allocs, 4,677 frees, 1,030,320 bytes allocated
==22334== 
==22334== All heap blocks were freed -- no leaks are possible
==22334== 
==22334== For lists of detected and suppressed errors, rerun with: -s
==22334== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

File: src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.cpp
Comment (pereyra-m) [inline src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.cpp:1]: I've tested a CVE with no `x_remediations` field (no mandatory) and I got a seg. fault

**Review by pereyra-m — COMMENTED** [review]
Gj!

File: src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.cpp
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.cpp:1]: Thanks, good catch! I assume that the bridge class will validate this scenario and for that reason, I didn't add a check the state of that pointer. 

I'm now adding a test case and the validation.

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeRemediationsModel.hpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    *
+    * @note This function assumes a specific data structure in the 'Entry' object, including nested objects.
+    *       Ensure that the 'Entry' object conforms to the expected schema to avoid runtime errors.
+    *
+    * @see Entry - The data structure containing CVE and remediation information.
+    * @see RocksDBWrapper - The utility class for interacting with RocksDB databases.
+    */
+    static void updateRemediation(const Entry* data, Utils::RocksDBWrapper& m_remediationsDatabase)
+    {
+        try
```
Comment (MiguelazoDS) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeRemediationsModel.hpp:61]: What exception is trying to capture this try-catch block?

**Review by pereyra-m — APPROVED** [review]
LGTM!

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeRemediationsModel.hpp (hunk 1 above)
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeRemediationsModel.hpp:61]: The idea is to catch any possible error that could occur during the parsing, it's possibly very generic, but the behavior of any exception we can trait in the same way, indicating that X update could not be processed.

File: src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.cpp [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+                    {
+                        "url": "https://example.org/ESA-22-11-CVE-1337-1234"
+                    }
+                ],
+                "x_remediations": {
+                    "windows": [
+                        {
+                            "anyOf":["KBT-800","KBT-1000","KBT-3000"],
+                            "products": ["Windows 10 - HastaLaVistaBaby", "Windows 11 - RiseOfTheMachines", "Windows 12 - Genisys"],
+                            "type": "update"
```
[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.cpp:76]: I didn't find any real info, but the behavior and intention of the test wouldn't change. The only difference could be the amount of chars and a more boring test 😅

File: src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.cpp [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+                        "value": "OS Command Injection vulnerability parseFilename function of example.php in the Web Management Interface of Example.org Example Enterprise on Windows, MacOS and XT-4500 allows remote unauthenticated attackers to escalate privileges.\n\nThis issue affects:\n  *  1.0 versions before 1.0.6\n  *  2.1 versions from 2.16 until 2.1.9."
+                    }
+                ],
+                "references": [
+                    {
+                        "url": "https://example.org/ESA-22-11-CVE-1337-1234"
+                    }
+                ],
+                "x_remediations": {
+                    "alpine": [
```
[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.cpp:136]: We should remove it from the `cve5` schema. 
I put @Dwordcito in the copy to confirm and attach this change in this PR if necessary.

File: src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.cpp [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+                        {
+                            "anyOf":["KBT-800","KBT-1000","KBT-3000"],
+                            "products": ["Windows 10 - HastaLaVistaBaby", "Windows 11 - RiseOfTheMachines", "Windows 12 - Genisys"],
+                            "type": "update"
```
[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.cpp:140]: Ditto

File: src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.cpp [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+                "references": [
+                    {
+                        "url": "https://example.org/ESA-22-11-CVE-1337-1234"
+                    }
+                ],
+                "x_remediations": {
+                    "windows": [
+                        {
+                            "products": ["Windows 10 - HastaLaVistaBaby", "Windows 11 - RiseOfTheMachines", "Windows 12 - Genisys"],
+                            "type": "update"
```
[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.cpp:203]: Ditto

File: src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.h [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+{
+protected:
+    /**
+     * @brief Construct a new Database Feed Manager Tests object
+     *
+     */
+    StoreRemediationsModelTest() = default;
+
+    /**
+     * @brief Destroy the Database Feed Manager Tests object
```
[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.h:32]: Good catch. Layer 8 error in recycle!

File: src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.cpp:136]: We still can avoid creating an example with Alpine Linux, later in another issue we can fix the cve5 schema.

[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/storeRemediationsModel_test.cpp:136]: If I remove it, the parse of the schema fails. The idea is to use any other valid field del schema that not be `windows`.

**Review by MiguelazoDS — APPROVED** [review]

**Pull request merged**